build.zig

```diff
@@ -639,10 +644,15 @@ fn addStaticLlvmOptionsToExe(exe: *std.build.LibExeObjStep) !void {
     }
 
     exe.linkSystemLibrary("z");
-    exe.linkSystemLibrary("zstd");
```

Comment:
ls `zstd` not supported when targeting `windows-msvc`?

Reply:
I just checked this again, and I think removing this conditionally was actually unnecessary. The key part of the zstd change was adding the `ZIG_ENABLE_ZSTD` flag to the cmake script - since the initial LLVM built during bootstrapping on the host is build without ZSTD enabled (since we haven't built it for the host yet), we don't want to require it during the build:

```
if(ZIG_STATIC_ZSTD AND ZIG_ENABLE_ZSTD)
    list(REMOVE_ITEM LLVM_SYSTEM_LIBRARIES "-lzstd")
    find_library(ZSTD NAMES libzstd.a libzstdstatic.a zstd NAMES_PER_DIR)
    list(APPEND LLVM_LIBRARIES "${ZSTD}")
endif()
```

Will remove -Ddisable-zstd and the change to this function.